src/backend/distributed/metadata/node_metadata.c

```diff
@@ -757,6 +757,18 @@ master_update_node(PG_FUNCTION_ARGS)
 		if (force)
 		{
 			handle = StartLockAcquireHelperBackgroundWorker(MyProcPid, lock_cooldown);
+			if (!handle)
+			{
+				/*
+				 * set lock_timeout to lock_cooldown, because if there is no slot for
+				 * background worker, we don't want it to wait forever to get lock.
+				 */
+				StringInfo lockCooldownStringValue = makeStringInfo();
+				appendStringInfo(lockCooldownStringValue, "%d", lock_cooldown);
+				set_config_option("lock_timeout", lockCooldownStringValue->data,
```

Comment:
There are some open comments by Onur and Talha about wrapping this in a utility function.

Reply:
I am also saying that we can remove `lockCooldownStringValue` and use a utility method to convert an `int` to `char*`. Because that is what `lockCooldownStringValue` is used for.